app/controllers/Checkout.scala

```diff
@@ -96,7 +98,10 @@ object Checkout extends Controller with LazyLogging with ActivityTracking {
 	      product.foreach(prod => trackAnon(SubscriptionRegistrationActivity(MemberData(result, formData, prod))))
       }
 
-      Redirect(routes.Checkout.thankYou()).withSession(session)
+      Ok(Json.obj("redirect" -> routes.Checkout.thankYou().url)).withSession(session);
+    }.recover {
+      case err: soap.Error if err.code == "TRANSACTION_FAILED" => Forbidden
```

Comment:
Yes, that's a good point. I will be doing separate PRs for improving this error handling in membership-common, membership-frontend and subscriptions-frontend.